sklearn/linear_model/tests/test_coordinate_descent.py

```diff
@@ -661,11 +661,6 @@ def test_check_input_false():
     clf = ElasticNet(selection='cyclic', tol=1e-8)
     # Check that no error is raised if data is provided in the right format
     clf.fit(X, y, check_input=False)
-    X = check_array(X, order='F', dtype='float32')
-    clf.fit(X, y, check_input=True)
```

Comment:
Because it casts X in 32 bits for the test below (row 668). But since you suggested changing the `ValueError` test to a smoke test, I'll put it back.